### ISSUE TITLE
 std::this_thread::sync_wait should not be pipeable

### ISSUE THREAD

**Issue by LeeHowes** [NVIDIA/stdexec#9]
We've found algorithms like this to be confusing in pipe syntax. It also doesn't return a sender.

6.1 example then would be:

> auto [result] = std::this_thread::sync_wait(execution::schedule(get_thread_pool())
>     | execution::then([]{ return 123; })
>     | execution::transfer(cuda::get_scheduler())
>     | execution::then([](int i){ return 123 * 5; })
>     | execution::transfer(get_thread_pool())
>     | execution::then([](int i){ return i - 5; }));

**Comment by mjgarland** [comment]
Is the proposed convention that only operators that return senders should be allowed in these pipelines?  Or is it only `sync_wait` itself that raises this concern?

**Comment by LeeHowes** [comment]
It's more of an algorithm-by-algorithm convention based on use conventions.

Sync_wait consumes a sender pipeline, and we have found it is much easier to read as a function call rather than a terminal pipeline node. 

on scopes a chunk of pipeline and applies up into it, rather than down a pipeline from it, so we have found it preferable for on to not be pipeable either - this is something we discussed earlier (Michal and I certainly went into some detail on it, at least) in the context of the argument order for on.
